lib/eventsource.js

```diff
@@ -471,3 +470,16 @@ function removeUnsafeHeaders (headers) {
 
   return safe
 }
+
+/**
+ * Transform an URL to a valid origin value.
+ *
+ * @param {String|Object} url URL to transform to it's origin.
+ * @returns {String} The origin.
+ * @api private
+ */
+function getOrigin (url) {
+  if (typeof url === 'string') url = parse(url)
+  if (!url.protocol || !url.hostname) return 'null'
```

Comment:
IMO `'null'` is a little confusing. It could be perceived as a bug? Maybe we should return `'invalid url'`?